### PR TITLE
Support kubectl deploy absolute manifest files

### LINKED ISSUE #2010: Deploy with kubectl support absolute manifest path
### Expected behavior
Deploy using kubectl support absolute manifest file path. Then we could run skaffold anywhere without considering the relative path of the manifest file.

### Actual behavior
Deploy using kubectl use a relative path to the directory where skaffold is running.

### Information

- Skaffold version: v0.27.0
- Operating system: Linux

### PR DESCRIPTION
**Pull request by tossmilestone**
Signed-off-by: Xiaoxi He <xxhe@alauda.io>

Fix #2010

### CODE REVIEW

**Comment by tossmilestone** [comment]
I signed it!

**Comment by tejal29** [comment]
@tossmilestone before we review it, i want to understand more about your usecase.
You are using `skaffold deploy` to deploy a manifest which is not part of your workspace, but references artifacts built by your workspace?

Thanks
Tejal

**Comment by tossmilestone** [comment]
@tejal29 Yes, I am using one skaffold yaml for several projects located in different directory. So it's useful if I can deploy using absolute file.

File: pkg/skaffold/util/util.go [hunk 1]
```diff
@@ -89,6 +89,12 @@ func StrSliceContains(sl []string, s string) bool {
 func ExpandPathsGlob(workingDir string, paths []string) ([]string, error) {
 	expandedPaths := make(map[string]bool)
 	for _, p := range paths {
+		if _, err := os.Stat(p); err == nil {
```
Comment (tejal29) [inline pkg/skaffold/util/util.go]: Can you instead use `AbsolutePaths` here?

Comment (tejal29) [reply pkg/skaffold/util/util.go]: https://github.com/GoogleContainerTools/skaffold/blob/782c111a2ab1347a99e07cc8093f090f2b5517d8/pkg/skaffold/util/util.go#L265

Comment (tossmilestone) [reply pkg/skaffold/util/util.go]: OK

Comment (tossmilestone) [reply pkg/skaffold/util/util.go]: I use `filepath.IsAbs`, cause the `AbsolutePaths` is used to prepend the relative paths.

File: pkg/skaffold/util/util.go [hunk 2]
```diff
@@ -89,6 +89,12 @@ func StrSliceContains(sl []string, s string) bool {
 func ExpandPathsGlob(workingDir string, paths []string) ([]string, error) {
 	expandedPaths := make(map[string]bool)
 	for _, p := range paths {
+		if filepath.IsAbs(p) {
+			// This is a absolute file reference
+			expandedPaths[p] = true
+			continue
```
Comment (corneliusweig) [inline pkg/skaffold/util/util.go]: this short-circuits the glob expansion further below. Should it be that way?

**Pull request merged**